Firestore/core/src/remote/firebase_platform_logging.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_REMOTE_FIREBASE_PLATFORM_LOGGING_H_
+#define FIRESTORE_CORE_SRC_REMOTE_FIREBASE_PLATFORM_LOGGING_H_
+
+#include <string>
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+/**
+ * Wraps the platform-dependent functionality associated with Firebase platform
+ * logging.
+ */
+class FirebasePlatformLogging {
```

Comment:
Consider renaming this to a regular noun rather than a gerund (i.e. "logger" vs "logging").

Reply:
In a forest, the person who cuts the trees is a "logger", while the larger practice is "logging". They're already grammatically related without resorting to calling the lumberjack a "tree logging logger".

My general skepticism with the "firebase platform logging" name comes from the fact that the term names the larger effort including something client-side that sends data, server-side collection, logs that are written, and the warehousing scripts to bucket the data. Since this is just a part of that larger effort, it seems misleading to name this smaller part with the name of whole thing.

To be fully descriptive you might name this thing "firebase platform logging metadata provider", since this doesn't really log data so much as attach headers to a request (aka provide request metadata). I think that's probably more of a mouthful than is warranted, given this thing's narrow scope.

If you agree with my idea above about (eventually) partitioning the header-providing bits along the lines of Firebase/Cloud/Auth/etc, possibly it's just sufficient to call this a FirebaseMetadataProvider.

Thoughts?

Reply:
I treat the feature name as an opaque identifier, so I would see it as similar to "LCD display/PIN number/etc.". However, I like the name `FirebaseMetadataProvider`, so I might as well just go with that.

I fully agree with partitioning metadata providers into Cloud/Firebase/etc.